### DISCUSSION TITLE
Are DocFX dead or dying?

### DISCUSSION THREAD

**Discussion by OptionHenrik** [dotnet/docfx#8250] [Q&A]
Hi DocFX maintainers, 

It seems to me that the DocFX project is somewhat dying. V3 has been underway since 2001 and the roadmap.md hasn't been updated for a very long time.

Can anyone please shed some light on project status? Are Microsoft still using DocFX or have they moved on to something else? ... what, in that case?

DocFX has been a great tool and it would be sad to see it die...

All the best
/Henrik

**Comment by paulushub** [comment]
> It seems to me that the DocFX project is somewhat dying. 

I am not one of the DocFX maintainers, but there is nothing about DocFX showing a sign of dying. 
My personal view is that, it is one of those unusual products; used internally by Microsoft, but not all the supporting tools and utilities are needed by the open source version and will not be made public.

* DocFX 2 is still being updated, and PR are actively committed.
* DocFX 3 by the maintainers is the tool currently used to build and deploy the `learn.microsoft.com` documentations. However, creating a  version for the open source community is clearly challenging to them since the DocFX 3 is still constantly evolving, and some of the stuff are only for internal uses and designed for learn.microsoft.com. 

> V3 has been underway since 2001 and the roadmap.md hasn't been updated for a very long time.

I do not know about the **2001**, but the roadmap.md was first published in Mar 29, 2018.

> Can anyone please shed some light on project status? 

You can still use the DocFX 3, if you are willing to do a little of work. The conceptual works without any problem, but the API reference will require a bit of work, see this thread https://github.com/dotnet/docfx/discussions/8227

> DocFX has been a great tool and it would be sad to see it die...

DocFX 2 works, and currently being ported to .NET 6.0.

**Reply by OptionHenrik** [reply]
Hi Paulushub, 

Thank you for taking the time to reply. 

I think this is a good example where Microsoft could do something for the greater good of the society by open sourcing the actual tool being used to create Learn.Microsoft.com ... without jeopardizing their daily business. 

Documentation is a fundamental part of all development projects and currently no products in the Microsoft portfolio manages to provide a proper solution for that ... perhaps with the exception of GitHub pages but docs.microsoft.com is a far more accessible and user-friendly way to do it.

Kill the cat or give it a proper home :-)

**Reply by markjulmar** [reply]
Hi Henrik, thanks for the discussion. DocFX is still used by learn.microsoft.com to parse Markdown and generate an object model we then create HTML from. It's not used exclusively however - there's a whole stream of systems around it (services, containers, etc.) that provide support to localize content, manage the publishing, synchronize repos, etc. DocFX is a small, but important tool in a much larger system.

Over the past few years, Microsoft Learn (FKA Docs) evolved into a much larger site with a bunch of dynamic features that ultimately drove us to a new architecture where we no longer use static site generation + a bunch of client-side JS, but have started moving to a dynamic composition layer that takes data assets (some of which are created by DocFX) and pieces them together into the final HTML representation. This architecture allows the site to continue to evolve with new dynamic features that can change based on environment conditions (our documentation is hosted in multiple places and needs to render differently based on location for example).

As such, the project owners will be merging some of the V3 branch changes back to main (V2), but not all -- some of the changes were more experimental or specific to Learn which doesn't make sense to keep. Watch for a published roadmap soon - key contributors are working on that now. There are several fun things we'd like to do to DocFX that are more community oriented!

**Comment by lanthonyneville** [comment]
Can anyone say why the project is now read only?

**Comment by paulushub** [comment]
> Can anyone say why the project is now read only?

Which project? I have tried viewing the DocFX project in several browsers, and it is not read-only!

**Reply by lanthonyneville** [reply]
Earlier in the day it was marked read-only. Now it isn't.

**Reply by markjulmar** [reply]
That was a mistake :)
